m4/quiche.m4

```diff
@@ -58,7 +58,7 @@ if test "$has_quiche" != "0"; then
   quiche_have_libs=0
   if test "$quiche_base_dir" != "/usr"; then
     TS_ADDTO(CPPFLAGS, [-I${quiche_include}])
-    TS_ADDTO(LDFLAGS, [-L${quiche_ldflags} -rpath ${quiche_ldflags}])
+    TS_ADDTO(LDFLAGS, [-L${quiche_ldflags} -Wl,-rpath,${quiche_ldflags}])
```

Comment:
Although this way of passing rpath looks more familiar, why do we need to do this where we have `TS_ADDTO_RPATH` below?

Reply:
After a bit of googling, it seems like `-R`, which `TS_ADDTO_RPATH` adds, is not a common way. I feel like we should modify `TS_ADDTO_RPATH` to add `-Wl,-rpath` instead of `-R` (or we can add the both if needed).

In any case, this actually solves the weird link issue I faced (https://github.com/apache/trafficserver/issues/10061).

Reply:
yes, makes sense, I think we can land this and then I will update the `common.m4` to do it right. https://github.com/apache/trafficserver/pull/10096